TLM/TLM/Util/TaskList.cs

```diff
@@ -0,0 +1,241 @@
+namespace TrafficManager.Util {
+
+    /// <summary>
+    /// This list is designed specifically for persistent render tasks with
+    /// intermittent bulk additions/removals/updates due to camera move or
+    /// tool state change.
+    /// </summary>
+    /// <typeparam name="T">The type of render task to store.</typeparam>
+    /// <remarks><i>Delegates must be assigned prior to use!</i></remarks>
+    public class TaskList<T> {
+
+        /// <summary>
+        /// When arrays are initialsied or their capacity needs increasing,
+        /// they will be sized in multiples of this value. Must be Power of 2.
+        /// </summary>
+        public const int BLOCK_SIZE = 64;
```

Comment:
Here and everywhere else in this PR (also in many locations in our code)
Why do we use signed integers for sizes and indices?
Could be uint, or ulong, or ushort, but why int? Are negative values allowed?

Reply:
`FastList<T>` has `m_size` field and indexer is operating on `int` also `Array.Length` (`int[]`) is `int`. IndexOf returns -1 if item has not been found in the array also index arithmetic can cause errors (because of overflow):
![image](https://user-images.githubusercontent.com/19638970/166110894-f5402f7b-69a0-41ad-80ed-2d8840f15fcf.png)

Not to mention about the case when you use e.g. `ushort` as index and then create an array with `ushort.MaxValue` -> you **have to** loop using `int` index otherwise it will create infinite loop